scripts/nbconverted/explore_input_rnaseq.py

```diff
@@ -0,0 +1,85 @@
+# Use pearson correlation score to compare PA1673 profile with all other genes
+# Pearson correlation evaluates the linear relationship between two continuous variables
+data_corr = data_sorted.corr(method='pearson')
+data_corr.head()
+
+
+# In[7]:
+
+
+# Plot distribution of correlation scores
+data_corr_PA1673 = data_corr[gene_id]
+sns.distplot(data_corr_PA1673)
+
+
+# In[8]:
+
+
+get_ipython().run_cell_magic('time', '', 'from statsmodels.nonparametric.smoothers_lowess import lowess\n## 4.  Plot the expression of genes along the PA1673 gradient:\n# Are the genes changing linearly with respect to PA1673?\n\n# Sort samples by PA1673 expression (lowest --> highest)\ndata_sorted = data.sort_values(by=[gene_id])\ndata_sorted.head()\n\n# For each gene plot gene expression trend along PA1673 gradient\n\n# Initialize the figure\nplt.style.use(\'seaborn-darkgrid\')\n \n# create a color palette\npalette = plt.get_cmap(\'Set1\')\n \n# multiple line plot\nnum_genes = data_sorted.shape[1]\nnum_panels = 100\n\n# Output file directory\nbase_dir = os.path.join(os.path.dirname(os.getcwd()), "exploration_results", "real_data")\nos.makedirs(base_dir, exist_ok=True)\n    \nfor panel in range(mt.ceil(num_genes%num_panels)):\n    rows = mt.sqrt(num_panels)\n    cols = rows\n    num=0\n    \n    # Not incrementing correctly here\n    data_subsample = data_sorted.drop(gene_id, axis=1).iloc[: , (panel*num_panels):((panel+1)*num_panels)-1]\n    \n    for column in data_subsample:\n        num+=1\n\n        # Find the right spot on the plot\n        plt.subplot(rows,cols, num)\n\n        # Plot the lineplot --Add smoothing to see trend\n        y_smooth = lowess(data_sorted[gene_id], data_subsample[column])[:,1]\n        plt.plot(data_sorted[gene_id], y_smooth, marker=\'\', color=palette(num), linewidth=1.9, alpha=0.9, label=column)\n               \n        # Same limits for everybody!\n        plt.xlim(0,1)\n        plt.ylim(0,1)\n\n        plt.tick_params(labelbottom=\'off\')\n        plt.tick_params(labelleft=\'off\')\n        # Not ticks everywhere\n        #if num in range(7) :\n        #    plt.tick_params(labelbottom=\'off\')\n        #if num not in [1,4,7] :\n        #    plt.tick_params(labelleft=\'off\')\n\n        # Add title\n        plt.title(column, loc=\'left\', fontsize=5, fontweight=0, color=palette(num) )\n\n    # general title\n    plt.suptitle("How gene expression changed\\nalong gene {} gradient?".format(gene_id), fontsize=13, fontweight=0, color=\'black\', style=\'italic\', y=1.02)\n\n    # Axis title\n    plt.text(0.5, 0.02, \'Gene {} expression\'.format(gene_id), ha=\'center\', va=\'center\')\n    plt.text(0.06, 0.5, \'Gene i expression\', ha=\'center\', va=\'center\', rotation=\'vertical\')\n    \n    # Save each panel as a figure\n    file_out = PdfPages(os.path.join(base_dir, \'Panel_{}.pdf\'.format(panel)))\n    plt.savefig(file_out, format=\'pdf\', bbox_inches = \'tight\')\n    plt.show()\n    file_out.close()')
```

Comment:
imports should go in the first code cell

Probably don't need to time the whole execution, and just the `for panel in range(...)` bit, right?

Reply:
you have a note that it is not incrementing correctly? Is it now?

Reply:
I see the purpose of this now! We are searching for non linear relationships? Pretty cool. We do see a bunch! There are a lot of wild patterns.

Is there a better way to search for non linear patterns than just by eye?

Reply:
Also, it looks like the axis labels on these plots are a bit wonky

Reply:
1. Moved the import statement
2.  Yes the note was when I initially wrote the script and I realized at the end it wasn't indexing properly but it should be now
3.  When looking for nonlinear patterns, the only think I could think to do is 1) Do a correlation analysis to see which genes were Linearly correlated vs not and 2) To visualize them all because the ones that are not linearly correlated may mostly be noise but there may be some with interesting patterns and those are the ones we care about.  This is something I'll need to think a bit more about -- maybe doing some sort of smoothing and somehow quantifying the trend rule out the "noisy" patterns but need to think about the actual implementation here.

Reply:
ah, gotcha. Cool.

Could you test all relationships for a series of correlations? Maybe a Pearson and Spearman correlation plus some sort of non linear curve fitting. Maybe some combinations of values for these three scores could help prioritize which gene-gene relationships to focus on?

Reply:
Very good idea.  Added as issue: https://github.com/greenelab/Pseudomonas_latent_spaces/issues/12